flax/core/scope.py

```diff
@@ -387,7 +387,8 @@ def __init__(self,
                name: Optional[str] = None,
                mutable: CollectionFilter = False,
                parent: Optional['Scope'] = None,
-               path: Iterable[str] = ()):
+               path: Iterable[str] = (),
+               flags: Optional[Set] = None):
```

Comment:
Maybe the signature could be 

```python
flags: Sequence[str] = ()
```
and have
```
self.flags = set(flags)
```
This way `self.flags` is always a `set`.

Reply:
did something similar